torchrl/envs/common.py

```diff
@@ -218,10 +228,19 @@ def __init__(
             self.batch_size = torch.Size([])
 
     @classmethod
-    def __new__(cls, *args, **kwargs):
+    def __new__(cls, *args, _batch_locked=True, **kwargs):
         cls._inplace_update = True
+        cls._batch_locked = _batch_locked
         return super().__new__(cls)
 
+    @property
+    def batch_locked(self) -> bool:
+        return self._batch_locked
```

Comment:
we should document what batch_locked means here. A proper docstring would be nice